svc.go

```diff
@@ -51,9 +52,21 @@ func NewAutoNATService(ctx context.Context, h host.Host, opts ...libp2p.Option)
 		dialer: dialer,
 		reqs:   make(map[peer.ID]int),
 	}
-	h.SetStreamHandler(autonat.AutoNATProto, as.handleStream)
 
-	go as.resetRateLimiter()
+	s, err := h.EventBus().Subscribe(&event.EvtLocalRoutabilityPublic{})
```

Comment:
@willscott @Stebalien 

1. Does this mean that every AutoNat server will also have to be an AutoNat client () ? If yes, who "starts" the client on the server ?
2. From the linkedin issue
```
This isn't perfect as one node could have configured port forwarding but it should help.
```

How can port forwarding spoil the party here ?

Reply:
> every AutoNat server will also have to be an AutoNat client

Yes. If we intend that nodes offering AutoNat Service are publicly reachable and valid DHT participants, then they need to learn that.

> why is port forwarding problematic?

I think the scenarios with port forwarding we want to understand are:
* I tell my router "public 4001 -> local 4001", and then configure IPFS locally to be on port 4001. Outbound connections, even if sourced from, say, `192.168.0.1:4001`, will get NATed by my router to `publicIP:<ephemeralport>`. We should potentially dial back on `publicIP:<low ports seen with private IPs>` to identify these cases.

Reply:
>  If yes, who "starts" the client on the server?

I believe we'll just have to start it on all nodes a the libp2p layer. But this does raise an interesting issue with the events system: dependencies.

> I think the scenarios with port forwarding we want to understand are

Note: If we fix https://github.com/libp2p/go-libp2p-autonat/issues/47 and https://github.com/libp2p/go-libp2p-autonat-svc/issues/44 (which we can do by listening for NAT routability events), this should be a non-issue.